usecases/sharding/replicated_index.go

```diff
@@ -13,24 +13,36 @@ package sharding
 
 import (
 	"context"
+	"fmt"
 
 	"github.com/go-openapi/strfmt"
-	"github.com/pkg/errors"
+	"github.com/semi-technologies/weaviate/entities/replica"
 	"github.com/semi-technologies/weaviate/entities/schema"
 	"github.com/semi-technologies/weaviate/entities/storobj"
+	"github.com/semi-technologies/weaviate/usecases/objects"
 )
 
 type ReplicatedIndexFactory interface {
 	GetReplicatedIndex(className schema.ClassName) Replicator
 }
 
 type Replicator interface {
-	PutObject(ctx context.Context, shardName string,
-		obj *storobj.Object) error
-	DeleteObject(ctx context.Context, shardName string,
-		id strfmt.UUID) error
-	BatchPutObjects(ctx context.Context, shardName string,
-		objs []*storobj.Object) []error
+	ReplicateObject(ctx context.Context, shardName, requestID string,
+		object *storobj.Object) replica.SimpleResponse
+	ReplicateObjects(ctx context.Context, shardName, requestID string,
+		objects []*storobj.Object) replica.SimpleResponse
+	ReplicateUpdate(ctx context.Context, shard, requestID string,
+		doc *objects.MergeDocument) replica.SimpleResponse
+	ReplicateDeletion(ctx context.Context, shardName, requestID string,
+		uuid strfmt.UUID) replica.SimpleResponse
+	ReplicateDeletions(ctx context.Context, shardName, requestID string,
+		docIDs []uint64, dryRun bool) replica.SimpleResponse
+	ReplicateReferences(ctx context.Context, shard, requestID string,
+		refs []objects.BatchReference) replica.SimpleResponse
+	CommitReplication(ctx context.Context, shard,
+		requestID string) interface{}
+	AbortReplication(ctx context.Context, shardName,
+		requestID string) interface{}
 }
 
 type ReplicatedIndex struct {
```

Comment:
ReplicatedIndex is not directly related to replication hence we can safely move it to another location.
I think a better place would be `repos/db` where the Replicator can be implement by `DB type` in replication.go

Reply:
Ok, moved